source/MaterialXContrib/MaterialXNode/MaterialXShadingNodeImpl.cpp

```diff
@@ -79,27 +81,58 @@ MStatus bindFileTexture(MHWRender::MShaderInstance& shader,
     MHWRender::MTextureManager* textureManager = renderer ? textureManager = renderer->getTextureManager() : nullptr;
     if (textureManager)
     {
-        MaterialX::FilePath imagePath = MaterialXMaya::findInSubdirectories(searchPath, fileName);
-        if (!imagePath.isEmpty())
+        MTexture *texture = nullptr;
+        if (udimSet && !udimSet->empty())
         {
+            std::vector<int> udimTiles;
+            mx::FilePathVec udimPaths = mx::ImageHandler::getResolveUDIMInformation(fileName, *udimSet, udimTiles);
+            if (udimTiles.size() != udimPaths.size() * 2)
             {
-                VP2TextureUniquePtr texture(textureManager->acquireTexture(imagePath.asString().c_str(), MaterialX::EMPTY_STRING.c_str()));
-                if (texture)
-                {
-                    MHWRender::MTextureAssignment textureAssignment;
-                    textureAssignment.texture = texture.get();
-                    status = shader.setParameter(parameterName.c_str(), textureAssignment);
-
-                    // Get back the texture description
-                    texture->textureDescription(textureDescription);
-                }
+                throw mx::Exception("Failed to resolve UDIM information properly for file: " + fileName);
             }
+            MStringArray mTilePaths;
+            MFloatArray mTilePositions;
+            MColor undefinedColor;
+            MStringArray failedTilePaths;
+            MFloatArray uvScaleOffset;
+            for (size_t i=0; i<udimPaths.size(); i++)
+            {
+                MaterialX::FilePath resolvedPath = MaterialXMaya::findInSubdirectories(searchPath, udimPaths[i]);
+                const mx::FilePath& udimPath = resolvedPath;
+                mTilePaths.append(udimPath.asString().c_str());
+                mTilePositions.append(static_cast<float>(udimTiles[i * 2]));
+                mTilePositions.append(static_cast<float>(udimTiles[i * 2 + 1]));
+            }
+            const unsigned int UDIM_BAKE_WIDTH = 4096;
+            const unsigned int UDIM_BAKE_HEIGHT = 4096;
+            // Note: we do not use the uv scale and offset. Ideally this should be used for the texture lookup code
+            // but at this point the shader code has already been generated.
+            texture = textureManager->acquireTiledTexture(fileName.c_str(), mTilePaths, mTilePositions, 
```

Comment:
This should ideally be wrapped in the unique_ptr immediately, to exclude any possibility of MaterialX throwing before it's wrapped.

Reply:
I would think that this wrapper, when it goes out of scope, should free what it holds delete so you don't have to care if something throws? If not, let me know how to write this as it would be a variable within the scope of an if() so how do you use it later on?
BTW, the next line wraps it so hard to have it fail as is.